src/conformal_models/inductive_bayes_regression.jl

```diff
@@ -0,0 +1,111 @@
+#using LaplaceRedux.LaplaceRegression
+using LaplaceRedux: LaplaceRegression
+
+@doc raw"""
+The `BayesRegressor` is the simplest approach to Inductive Conformalized Bayes. As explained in https://arxiv.org/abs/2107.07511,
+the  conformal score is  defined as the opposite of the probability of observing y given x : `` s= -P(Y|X) ``. Once the treshold ``\hat{q}`` is chosen, The credible interval is then
+ computed as the range of y values so that 
+ `` C(x)= \big\{y : P(Y|X) > -\hat{q} \big} ``
+"""
+mutable struct BayesRegressor{Model<:Supervised} <: ConformalInterval
+    model::Model
+    coverage::AbstractFloat
+    scores::Union{Nothing,AbstractArray}
+    heuristic::Function
+    train_ratio::AbstractFloat
+end
+
+function BayesRegressor(
+    model::Supervised;
+    coverage::AbstractFloat=0.95,
+    heuristic::Function=ConformalBayes,
+    train_ratio::AbstractFloat=0.5,
+)
+    @assert typeof(model) == LaplaceRegression "Model must be of type Laplace"
```

Comment:
Can also just type the model field instead of using the assert?

Reply:
Or perhaps indeed we should not assert the model type at all and then get rid of the LaplaceRedux dependency